xds/internal/balancer/clusterresolver/resource_resolver.go

```diff
@@ -222,12 +231,18 @@ func (rr *resourceResolver) stop() {
 	rr.childrenMap = make(map[discoveryMechanismKey]discoveryMechanismAndResolver)
 	rr.mechanisms = nil
 	rr.children = nil
+
 	rr.mu.Unlock()
 
 	for _, r := range cm {
 		r.r.stop()
 	}
 
+	if closing {
+		rr.serializerCancel()
+		<-rr.serializer.Done()
+	}
```

Comment:
Is there a way to close the serializer without causing any queued unran callbacks to execute? I don't think there is, but should we add that method to that type? It feels like a waste to execute all the potentially operations (i.e. generateLocked()) when we know the whole cluster resolver component is closing anyway.

Reply:
Yes, there is currently no way to cause the serializer to shut down without running queued callbacks. In fact, the serializer in its first version used to do that, but when I was making changes for channel idleness, I quickly realized that the more common case was the one where it is guaranteed that all scheduled callbacks are run.

It should be trivial for the user of the serializer to have some logic to turn the callback into a no-op when it knows that the context passed to the serializer has been cancelled.

Reply:
Ok. So you think that say if you have 5 generateLocked() calls queued, it's fine to just run them before closing, which would search through all discovery mechanisms. Eh, whatever, minor, in practice won't scale out of proportion.